### ISSUE TITLE
Make Squirrel not launch program after install

### ISSUE THREAD

**Issue by upn0rth** [Squirrel/Squirrel.Windows#983]
I was wondering if there are any way not having the program launch after the install. The reason being that some files probably arent loaded into the first run of the program when using the Squirrel installer. This causes my program to crash. All other launches of the program are working just fine, but not the first run.

Thanks

**Comment by Andrew-Hanlon** [comment]
You should re-read the documentation around [Custom Squirrel Events](https://github.com/Squirrel/Squirrel.Windows/blob/master/docs/using/custom-squirrel-events.md).

**Comment by upn0rth** [comment]
Yep, that's what I have been looking at. Is there any wiki list of the functionality of the custom events? Or is these the only docs? https://github.com/Squirrel/Squirrel.Windows/tree/master/docs/using

**Comment by Slion** [comment]
You could use onFirstRun to set a boolean much like it's done in that example and later test that boolean to exit the program.

**Comment by Thieum** [comment]
@upn0rth did you try the solution provided by @Slion ? If it fixed your issue, you can close this.

**Comment by varKeytrap** [comment]
If anyone is looking for the solution : use the `--silent` switch on the setup.
E.g : `setup.exe --silent`

**Comment by cope** [comment]
@keytrap-x86 How do we make the user add `--silent` when they download and click the installer?

**Comment by RareScrap** [comment]
@cope You can run install via command line like:
```powershell
& '.\MyAppSquirrelInstaller.exe' --silent
```

**Comment by keytrap-x86** [comment]
Kind if awkward question because *It depends™*, we can't control what the user will do nor force him to add it but you could create a "fake lightweight installer" which would only download the "real" setup.exe and start it adding the `--silent` arg.
A lot of editors do that, they make you download the little installer and  you think "Wow that was fast", start it and have a screen saying that it's downloading the "real" soft

**Comment by anaisbetts** [comment]
> You should re-read the documentation around [Custom Squirrel Events](https://github.com/Squirrel/Squirrel.Windows/blob/master/docs/using/custom-squirrel-events.md).

^^ Good advice! Squirrel tells you when the launch is the first launch. You could just exit!
